ContentSerializer/ContentConfiguration.cs

```diff
@@ -1,94 +1,110 @@
             {
-                get
-                {
-                    return new Regex[] {
-                        new Regex(@"^DATA\/Spells\/[^\/]*\.inibin\z"),
-                        new Regex(@"^DATA\/Shared\/Spells\/[^\/]*\.inibin\z"),
-                        new Regex(@"^DATA\/Characters\/.*\/Spells\/[^\/]*\.inibin\z")
-                    };
-                }
+                if (!content.Values.ContainsKey(field.Item1))
+                    continue;
+                if (!content.Values[field.Item1].ContainsKey(field.Item2))
+                    continue;
+                var nameKey = content.Values[field.Item1][field.Item2].ToString();
+                if (string.IsNullOrEmpty(nameKey))
+                    continue;
+                if (!Localization.Content.ContainsKey(nameKey))
+                    continue;
+                return Localization.Content[nameKey];
             }
-
-            public Spell(FontConfigFile localization) : base(localization) { }
+            return name;
         }
 
-        public class Item : ContentConfiguration
+        public static List<Regex> LoadPatterns(string file)
         {
-            public override string ConversionMapPath { get { return "ConversionMaps/ItemConversionMap.json"; } }
-            public override string OutputDirectory { get { return "Items"; } }
-            public override Regex[] ResourcePatterns
+            var data = File.ReadAllText(file);
+            var patterns = JsonConvert.DeserializeObject<List<string>>(data);
+            var patternsRegex = new List<Regex>();
+            foreach (var pattern in patterns)
             {
-                get
-                {
-                    return new Regex[] {
-                        new Regex(@"^DATA\/Items\/[^\/]*\.inibin\z")
-                    };
-                }
+                patternsRegex.Add(new Regex(pattern, RegexOptions.IgnoreCase));
             }
+            return patternsRegex;
+        }
 
-            public Item(FontConfigFile localization) : base(localization) { }
+        public static ContentConfiguration Load(FontConfigFile localization, string patternsFile,
+            string mapFile, string InputPattern, string outputPattern)
+        {
+            var converter = new InibinConverter(ConversionMap.Load(mapFile));
+            var patterns = LoadPatterns(patternsFile);
+            return new ContentConfiguration(localization, converter, patterns,
+                new Regex(InputPattern,RegexOptions.IgnoreCase), outputPattern);
+        }
 
-            public override object FindId(string archiveFilePath)
+        public static ContentConfiguration[] LoadList(FontConfigFile localization,
+            string mapDir, string patternDir, string listFile)
+        {
+            var list = new List<ContentConfiguration>();
+            var data = File.ReadAllText(listFile);
+            var entryList = JsonConvert.DeserializeObject<List<ContentConfigurationListEntry>>(data);
```

Comment:
Instead of having a `ContentConfigurationListEntry` class, you could directly serialize/deserialize a `List<ContentConfiguration>`. Would also simplify the loading a lot.